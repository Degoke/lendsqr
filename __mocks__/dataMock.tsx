import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
