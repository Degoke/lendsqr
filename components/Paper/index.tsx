import { ReactNode } from 'react'
import styles from './Paper.module.scss'

type Props = {
  children: ReactNode
  className?: string
}

export default function Paper({ children, className }: Props) {
  return <div className={`${styles.paper} ${className}`}>{children}</div>
}
