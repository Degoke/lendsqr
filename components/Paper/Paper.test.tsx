import { render, screen } from '@testing-library/react'
import Paper from './index'
import { mockCard, MockComponent, mockImage } from '../../__mocks__/dataMock'

describe('Paper tests', () => {
  it('renders Paper component unchanged', () => {
    const { container } = render(
      <Paper>
        <MockComponent />
      </Paper>
    )
    expect(container).toMatchSnapshot()
  })
})
