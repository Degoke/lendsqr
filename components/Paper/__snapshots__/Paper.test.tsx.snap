// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper tests renders Paper component unchanged 1`] = `
<div>
  <div
    class="paper undefined"
  >
    <div
      data-testid="mockComponent"
    >
      Hello World
    </div>
  </div>
</div>
`;
